build/media-suite_compile.sh

```diff
@@ -2432,27 +2438,107 @@ if [[ $mpv != n ]] && pc_exists libavcodec libavformat libswscale libavfilter; t
+            # Mac-only options, we might as well catch and squash them
+            elif [[ $option =~ ^--enable-(macos-*|swift-*) ]]; then
+                meson_opts+=("${option/--enable-/-D}=disabled")
+                do_simple_print "${orange}Disabling option '${option}' as it's not usable on Windows"'!'"${reset}"
+            # Manpages corner
+            elif [[ $option =~ ^--enable-(html-build|manpage-build) ]]; then
+                meson_opts+=("${option/--enable-/-D}=enabled")
+            elif [[ $option =~ ^--disable-(html-build|manpage-build) ]]; then
+                meson_opts+=("${option/--disable-/-D}=disabled")
+            elif [[ $option =~ ^--enable-pdf-build ]]; then
+                meson_opts+=("${option/--disable-/-D}=disabled")
+                do_simple_print "${orange}Disabling PDF manual, currently broken"'!'"${reset}"
+            elif [[ $option =~ ^--lua=lua* ]]; then
+                # `meson setup build` does not take kindly to a format not matching lua(jit|\d\.\d)
+                local luaver="${option/--lua=lua}"
+                if [[ luaver =~ ^(5|-5) ]]; then
+                    luaver='5.${luaver:-1}'
+                elif [[ luaver =~ ^jit ]]; then
+                    luaver='jit'
+                fi
+                meson_opts+=("-Dlua=lua${luaver}")
+                # Remove the `-Dlua=enabled` flag if present, since the order of flags matters.
+                # - `-Dlua=lua5.2 -Dlua=enabled` makes Meson behave like only `-Dlua=enabled` was specified;
+                # - `-Dlua=enabled -Dlua=lua5.2` makes Meson take lua5.2 if available, erroring otherwise;
+                meson_opts=("${meson_opts[@]/ -Dlua=enabled}")
+            elif [[ $option =~ ^--enable-lua ]] && ! [[ "${meson_opts[@]}" =~ -Dlua= ]]; then
+                meson_opts+=("-Dlua=enabled")
+            fi
+            unset option
+        done
+        do_simple_print "${meson_opts[@]}"
+
         extra_script pre configure
-        CFLAGS+=" ${mpv_cflags[*]} -Wno-int-conversion" LDFLAGS+=" ${mpv_ldflags[*]}" \
+        local old_ldf="$LDFLAGS"
+        CFLAGS+=" ${mpv_cflags[*]} -Wno-int-conversion" LDFLAGS+=" ${mpv_ldflags[*]} -larchive -lb2 -lzstd" \
             RST2MAN="${MINGW_PREFIX}/bin/rst2man" \
             RST2HTML="${MINGW_PREFIX}/bin/rst2html" \
-            RST2PDF="${MINGW_PREFIX}/bin/rst2pdf2" \
-            PKG_CONFIG="$LOCALDESTDIR/bin/ab-pkg-config" \
-            log configure /usr/bin/python waf configure \
-            "--prefix=$LOCALDESTDIR" "--bindir=$LOCALDESTDIR/bin-video" \
-            "${MPV_OPTS[@]}"
+            RST2PDF="${MINGW_PREFIX}/bin/rst2pdf2"
+        mkdir build
+        PKG_CONFIG="pkgconf --keep-system-libs --keep-system-cflags" CC=${CC/ccache /}.bat CXX=${CXX/ccache /}.bat \
+            log "meson.setup" meson setup build "${meson_opts[@]}" --prefix="${LOCALDESTDIR}" --bindir="${LOCALDESTDIR}/bin-video" --default-library="${default_lib}" --buildtype=release
```

Comment:
log names are non-standard atm, but I expect there to be more changes required to the Meson calls.
Kept this order for now so that it
1) doesn't break existing `mpv_extra.sh` setups
2) I wanted it to work first, and then adapt it to use the helper functions later